library/core/src/cell.rs

```diff
@@ -1811,6 +1811,50 @@ impl<T: ?Sized + fmt::Display> fmt::Display for RefMut<'_, T> {
 ///
 /// [`.get_mut()`]: `UnsafeCell::get_mut`
 ///
+/// `UnsafeCell<T>` has the same in-memory representation as its inner type `T`. A consequence
+/// of this guarantee is that it is possible to convert between `T` and `UnsafeCell<T>`.
```

Comment:
This should include a example of the desired usage: foreign `*mut T` to `&UnsafeCell<T>`.